### PR TITLE
Support dependencies with multiple integrity values

### LINKED ISSUE #34: Errors when dependencies specify multiple integrity values
We are seeing an issue when a `package-lock.json` specifies dependencies with multiple integrity values:

```
Files could not be expanded: service error: ValueError: too many values to unpack (expected 2)

service node_modules failed:
time="2023-09-18T16:21:19+02:00" level=warning msg="Path \"/etc/SUSEConnect\" from \"/etc/containers/mounts.conf\" doesn't exist, skipping"
time="2023-09-18T16:21:19+02:00" level=warning msg="Path \"/etc/zypp/credentials.d/SCCcredentials\" from \"/etc/containers/mounts.conf\" doesn't exist, skipping"
Running /usr/lib/obs/service//node_modules --cpio node_modules.obscpio --output node_modules.spec.inc --source-offset 10000 --outdir /var/cache/obs/VJvo7VsTt4d9/out
Traceback (most recent call last):
File "/usr/lib/obs/service//node_modules", line 624, in
sys.exit(main(args))
File "/usr/lib/obs/service//node_modules", line 376, in main
process_packagelock_file(js)
File "/usr/lib/obs/service//node_modules", line 341, in process_packagelock_file
collect_v3_deps(js["packages"])
File "/usr/lib/obs/service//node_modules", line 326, in collect_v3_deps
add_standard_dependency(parse_supported_fetch_url(url), integrity, module, path)
File "/usr/lib/obs/service//node_modules", line 230, in add_standard_dependency
algo, chksum = integrity.split("-", 2)
ValueError: too many values to unpack (expected 2)
```

Here you have an example of a problematic case:

```json
    "node_modules/array-unique": {
      "version": "0.3.2",
      "resolved": "https://registry.npmjs.org/array-unique/-/array-unique-0.3.2.tgz",
      "integrity": "sha1-qJS3XUvE9s1nnvMkSp/Y9Gri1Cg= sha512-SleRWjh9JUud2wH1hPs9rZBZ33H6T9HOiL0uwGnGx9FpE6wKGyfWugmbkEOIs6qWrZhg0LWeLziLrEwQJhs5mQ==",
      "dev": true,
      "engines": {
        "node": ">=0.10.0"
      }
    },
```

As you can see, the `integrity` field contains the following values, which is causing the OBS service to fail:

* `sha1-qJS3XUvE9s1nnvMkSp/Y9Gri1Cg=`
* `sha512-SleRWjh9JUud2wH1hPs9rZBZ33H6T9HOiL0uwGnGx9FpE6wKGyfWugmbkEOIs6qWrZhg0LWeLziLrEwQJhs5mQ==`

### PR DESCRIPTION
**Pull request by marcosbc**
Fixes #34 

This PR fixes the scenario where a `package-lock.json` file may include dependencies with multiple integrity values, as it would cause an error on the OBS side when building. The solution involves selecting the first defined integrity and checking against that, and ignoring any other one.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,7 +23,8 @@ externally download sources for use by `npm` during `rpmbuild`.
 ## As OBS service
 
 - Get `package-lock.json` with `localfileVersion: 2`. For example,
-  - `npm install --package-lock-only --legacy-peer-deps` with npm 7+
+  - `npm install --package-lock-only --legacy-peer-deps --ignore-scripts`
+    with npm 7+
   - `--legacy-peer-deps` is required to fetch peer dependencies from remote
```
Comment (AdamMajer) [inline README.md]: Would it be possible to move this into another PR since it's unrelated here.

File: node_modules.py [hunk 2]
```diff
@@ -225,8 +225,9 @@ def make_unique_fn_from_path(o):
 
     return '-'.join(prepended + [original_fn])
 
-def add_standard_dependency(o, integrity, module, install_path):
+def add_standard_dependency(o, integrities, module, install_path):
     url = urllib.parse.urlunparse(o)
+    integrity = integrities.split(" ")[0]
```
Comment (AdamMajer) [inline node_modules.py]: would it make sense here to use the better of the multiple hashes here? For example, sha256 vs. sha1, pick sha256.

I guess this could be done in later PR if it's too much work here.

**Review by AdamMajer — CHANGES_REQUESTED** [review]
Looks OK to fix the problem but moving the README changes to different PR or at least different commit would be nice.

File: README.md (hunk 1 above)
Comment (marcosbc) [reply README.md]: Done!

File: node_modules.py (hunk 2 above)
Comment (marcosbc) [reply node_modules.py]: Good point. I have slightly modified this to pick the last element. From my observations, a `package-lock.json` will sort the integrity values from worse to better.

```
$ grep '"integrity": ".* ' package-lock.json
      "integrity": "sha1-7ihweulOEdK4J7y+UnC86n8+ce4= sha512-dRLjCWHYg4oaA77cxO64oO+7JwCwnIzkZPdrrC71jQmQtlhM556pwKo5bUzqvZndkVbeFLIIi+9TC40JNF5hNQ=="
      "integrity": "sha1-LMDWazHqIwNkWENuNiDYWVTGbDw= sha512-A48My/mtCklowHBlI8Fq2jFWK4tX4lJ5E6ytFsSOq1fzpvT0SQSgKhSg7lN5c2uYFOrUAOQp6zhhJnpp1eMloQ==",
      "integrity": "sha1-1PRWKwzjaW5BrFLQ4ALlemNdxtw= sha512-PhmXi5XmoyKw1Un4E+opM2KcsJInDvKyuOumcjjw3waw86ZNjHwVUOOWLc4bCzLdcKNaWBH9e99sbWzDQsVaYg==",
      "integrity": "sha1-u7aTucqRXCMlFbIosaArYJBD2+s= sha512-O2XLNDBIg1DnTOa+2XrIwSiXEV8h2KImXUnjhhn2+UsvZ+Es2uyd5CCRTNQlDGbzUQOW3aYCBx9rVA6dzsiY7Q==",
      "integrity": "sha1-J5siXfHVgrH1TmWt3UNS4Y+qBxM= sha512-R3f198pcvnB+5IpnBlRkphuE9n46WyVl8I39W/ZUTZLz4nqSP/oLYUrcnJrw462Ds8he4YKMov2efsTIw1BDGQ==",
      "integrity": "sha1-NakjLfo11wZ7TLLd8jV7GHFTbF8= sha512-lJxZYlT4DW/bRUtFh1MQIWqmLwQfAxnqWG4HhEdjMlkrJYnJn0Jrr2u3mgxqaWsdiBc76TYkTG/mhrnYTuzfHw==",
      "integrity": "sha1-NakjLfo11wZ7TLLd8jV7GHFTbF8= sha512-lJxZYlT4DW/bRUtFh1MQIWqmLwQfAxnqWG4HhEdjMlkrJYnJn0Jrr2u3mgxqaWsdiBc76TYkTG/mhrnYTuzfHw==",
      "integrity": "sha1-7GphrlZIDAw8skHJVhjiCJL5Zyo= sha512-nSVgobk4rv61R9PUSDtYt7mPVB2olxNR5RWJcAsH676/ef11bUZwvu7+RGYrYauVdDPcO519v68wRhXQtxsV9w==",
      "integrity": "sha1-TkkvjQTftviQA1B/btvy1QHnxvQ= sha512-Xhj93RXbMSq8urNCUq4p9l0P6hnySJ/7YNRhYNug0bLOuii7pKO7xQFb5mx9xZXWCar88pLPb805PvUkwrLZpQ==",
      "integrity": "sha1-hX/Kv8M5fSYluCKCYuhqp6ARsF0= sha512-miQKw5Hv4NS1Psg2517mV4e4dYNaO3++hjAvLOAzKqZ61rH8NS1SK+vbfBWZ5PY/Me/bEWhUwqMghEW5Fb9T7Q==",
      "integrity": "sha1-MAV0OOrGz3+MR2fzhkjWaX11yQM= sha512-SknJC52obPfGQPnjIkXbmA6+5H15E+fR+E4iR2oQ3zzCLbd7/ONua69R/Gw7AgkTLsRG+r5fzksYwWe1AgTyWA==",
      "integrity": "sha1-NakjLfo11wZ7TLLd8jV7GHFTbF8= sha512-lJxZYlT4DW/bRUtFh1MQIWqmLwQfAxnqWG4HhEdjMlkrJYnJn0Jrr2u3mgxqaWsdiBc76TYkTG/mhrnYTuzfHw==",
...
      "integrity": "sha1-R0tQhlrzpJqcRlfwWs0UVFj3fYI= sha512-GojqklwG8gpzOVEVki5KudKNoq7MbbjYZCbyWzEz7tyPA7eleiE0+ePwOWQQRb5fm86rD3S8Tc0tSFf3AOv50w==",
      "integrity": "sha1-HBH5IY8HYImkfdUS+TxmmaaoHVI= sha512-ncTzHV7NvsQZkYe1DW7cbDLm0YpzHmZF5r/iyP3ZnQtMiJ+pjzisCiMNI+Sj+xQF5pXhSHxSB3uDbsBTzY/c2A=="
```

While it is not a definitive solution, I think it should solve most cases. WDYT?

Comment (AdamMajer) [reply node_modules.py]: for purposes of this exercise, it's probably valid approach. Another would be to assume it's all sha### and pick the largest one (no need to sort, just pick the largest in the list)

Comment (marcosbc) [reply node_modules.py]: Good idea! I have implemented it with `max(lst, key=len)`

**Comment by AdamMajer** [comment]
I've squashed these commits and merged them now. Thanks!
